### PR TITLE
Added options to server config

### PR DESCRIPTION
**Pull request by jasonsliu**
Rip merge issues

### CODE REVIEW

**Review by ryanvoong — CHANGES_REQUESTED** [review]
update the config to be like this https://docs.google.com/presentation/d/1fs607JwiHSqA7EFc33PdSraK4wDf00Klx3EU-zLRk8g/view#slide=id.g1a865bf17f_0_27 
we need to be able to set both the url path and the base path for each handler

File: server.h [hunk 1]
```diff
@@ -16,6 +16,8 @@ class Server {
 
   boost::asio::ip::tcp::acceptor acceptor_;
   int port_;
+  std::map<std::string, std::string> echoPathToRoot_;
```
Comment (ryanvoong) [inline server.h]: can you add a comment describing what are the map's keys and values

File: config_parser.h [hunk 2]
```diff
@@ -60,4 +62,27 @@ class NginxConfigParser {
   TokenType ParseToken(std::istream* input, std::string* value);
 };
 
+struct ServerInfo {
+  ServerInfo() : port(1880) {}
+  int port;
+  std::map<std::string, std::string> echoPathToRoot;
```
Comment (ryanvoong) [inline config_parser.h]: can you add a comment describing what are the map's keys and values

**Review by ryanvoong — CHANGES_REQUESTED** [review]
names of variables/methods should be snake_case, not camelCase

**Review by ryanvoong — APPROVED** [review]

**Review by ryanvoong — APPROVED** [review]
a little nitpicky, but see 'Struct Data Members' section here https://google.github.io/styleguide/cppguide.html#Variable_Names

**Comment by crystalhsieh** [comment]
please fix merge conflicts :)

**Pull request merged**